app/component/CustomizeSearch.js

```diff
@@ -264,7 +354,9 @@ class CustomizeSearch extends React.Component {
     </section>);
 
   getModes() {
-    if (this.context.location.query.modes) {
+    if (getCustomizedSettings().modes && !this.context.location.query.modes) {
+      return getCustomizedSettings().modes;
+    } else if (this.context.location.query.modes) {
       return decodeURI(this.context.location.query.modes).split(',');
```

Comment:
Swapping the order of if tests will make the logic simpler. No need for extra ` !this.context.location.query.modes`  test.